src/main/java/com/rock_mc/syn/api/Log.java

```diff
@@ -1,16 +1,75 @@
 package com.rock_mc.syn.api;
 
+import com.google.common.collect.Lists;
 import com.rock_mc.syn.Syn;
 import com.rock_mc.syn.command.CmdManager;
+import com.rock_mc.syn.db.EventLog;
+import com.rock_mc.syn.db.PluginPlayerInfo;
 import com.rock_mc.syn.log.Logger;
+import com.rock_mc.syn.utlis.Utils;
 import org.bukkit.entity.Player;
+import org.bukkit.util.StringUtil;
+
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 public class Log {
     private final static String commandName = CmdManager.LOG;
+    private final static SimpleDateFormat sdf = new SimpleDateFormat("yyyy/MM/dd HH:mm");
 
-    public static void run(Syn plugin, Logger logger, Player player, String[] args) {
+    public static void exec(Syn plugin, Logger logger, Player player, String[] args) {
         synchronized (Syn.apiLock) {
-            // TODO: syn log [time] [player] [page]
+            long[] times = Utils.parseTime(args);
+            List<String> players = Utils.parseUsers(args);
+
+            Timestamp start = null;
+            Timestamp end = null;
+            Integer page = null;
+            Integer rows = null;
+
+            LocalDateTime currentDateTime = LocalDateTime.now(ZoneId.of("UTC"));
+            Instant now = currentDateTime.atZone(ZoneId.of("Asia/Taipei")).toInstant();
```

Comment:
Can we detect the timezone? Or set it in config?

Reply:
To better handle when the server's default time zone is not UTC, replaced the code by
`Instant now = currentDateTime.atZone(TimeZone.getDefault().toZoneId()).toInstant();`